extension/contentfragment/bundle/src/test/java/com/adobe/cq/wcm/core/components/extension/contentfragment/internal/models/v1/ContentFragmentElementImplTest.java

```diff
@@ -0,0 +1,108 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.extension.contentfragment.internal.models.v1;
+
+import com.adobe.cq.dam.cfm.ContentElement;
+import com.adobe.cq.dam.cfm.DataType;
+import com.adobe.cq.dam.cfm.FragmentData;
+import com.adobe.cq.dam.cfm.content.FragmentRenderService;
+import com.adobe.cq.dam.cfm.converter.ContentTypeConverter;
+import com.adobe.cq.wcm.core.components.extension.contentfragment.models.ContentFragment;
+import org.apache.commons.io.IOUtils;
+import org.apache.sling.api.resource.Resource;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertNotNull;
+
+public class ContentFragmentElementImplTest {
+
+    @Test
+    public void testElementGetParagraphsNotText() {
+        ContentFragment.Element element = getMockedElement( null, "application/pdf", false);
+        String[] paragraphs = element.getParagraphs();
+        assertNull(paragraphs);
+    }
+
+    @Test
+    public void testElementGetParagraphsNotMultiValueNullContent() {
+        ContentFragment.Element element = getMockedElement( null,"text/html", false);
+        String[] paragraphs = element.getParagraphs();
+        assertNull(paragraphs);
+    }
+
+    @Test
+    public void testElementGetParagraphsNotMultiValueWithContent() {
+        ContentFragment.Element element = getMockedElement( "simple text","text/plain", false);
+        String[] paragraphs = element.getParagraphs();
+        assertNotNull(paragraphs);
+        assertEquals(1, paragraphs.length);
+    }
+
+    @Test
+    public void testStyledHtmlParagraphs() throws IOException {
+        ContentFragment.Element element = getMockedElement( getContent("sample_html_snippet.dat"),"text/html", false);
+        String[] paragraphs = element.getParagraphs();
+        assertNotNull(paragraphs);
+        assertEquals(10, paragraphs.length);
+    }
+
+    @Test
+    public void testPlainParagraphs() throws IOException {
+        ContentFragment.Element element = getMockedElement( getContent("sample_plain.dat"),"text/plain", false);
```

Comment:
This test doesn't make sense IMHO because of the way things are mocked.

In reality, RenderService would actually convert the plain text into HTML before parsing it. So the actual result should be 5.

As tests can also be considered documentation, I would remove it, as it provides a wrong impression of how things are supposed to work.

Reply:
ok